paddle/fluid/framework/details/build_strategy.cc

```diff
@@ -53,6 +58,17 @@ class ParallelExecutorPassBuilder : public ir::PassBuilder {
       }
     }
 
+    CollectiveContext *context = CollectiveContext::GetInstance();
+    context->end_points_ = strategy_.trainers_end_points_;
+    context->rank_id_ = strategy_.trainer_id_;
```

Comment:
rank is not trainer_id, a rank stands for a "worker", which also can be one GPU device.

Reply:
Done.